src/dense.jl

```diff
@@ -167,26 +167,38 @@ function next_interval!(ivp, integ, istate, step_prev, tout)
     return abort
 end
 
+
 """
+Makes dense output
+
+interpolate!{ST}(dstate::DenseState{ST}, tout, step_out)
+
+Input:
 
-Make dense output using Hermite interpolation of order O(3). Updates
-yout in-place.  Only needs y and dy at t1 and t2.
-Input
-- istate::AbstractState -- istate of the integrator at time t2
-- step_prev::Step -- solution at time t1 respectively
+- `dstate::DenseState{ST}`
+  - `ST` is the `AbstractIntegrator` state which can be used for dispatch
 - tout -- time of requested output
-- yout -- inplace y output
-Ref: Hairer & Wanner p.190
+- step_out::Step -- inplace output step
+
+Output: nothing
 
-TODO: tdir (I think this works for any order of t1 and t2 but needs
-verifying.
+TODO: output dy too
 
-TODO: fill dy
+TOOD: provide arbitrary order dense output. Maybe use work of @obiajulu on A-B-M methods.
+"""
+function interpolate! end
 
-TODO: arbitrary order method (change step_prev::Step to step_prevs::Tuple{Step,N})
+"""
+Make dense output using Hermite interpolation of order O(3), should
+work for most integrators and is used as default.  This only needs y
+and dy at t1 and t2.
 
+Ref: Hairer & Wanner p.190
 """
-function interpolate!{T,Y}(istate::AbstractState,step_prev::Step{T,Y},tout::T,step_out::Step{T,Y})
+function interpolate!(dstate::DenseState, tout, step_out::Step)
```

Comment:
As discussed, I changed this signature.  Dispatch for specialized methods can be done like:
```
function interpolate!{I<:RKIntegrator}(dstate::DenseState{I}, tout, step_out)
```
I think that was your concern.